IFC 4x3 integration (including .NET8 port)

andyward opened the pull request:
An initial pass integrating both the new GE and IFC4x3 support into xbim Xplorer

This necessitated upgrading to .net6 which was overdue

andyward committed:
First pass ifc4x3 implementation.

Supports loading IFC4x3 schema but latest Geometry needed

Styler fixed up and multi-schema code reviewed

andyward committed:
WIP upgrade to .net6.0 and latest netcore geometry
(including 4x3).
Geometry to test.
Plugins disabled due to Nuget package deprecation

andyward committed:
Integrated Essentials fix to allow 4x3 geometry to be created

andyward committed:
Fixed up Tests, Winforms and Plugin projects
Tidy up of projects

andyward commented:
@SteveLockley , @martin1cerny  if you're interested in the current state of the GE6

andyward commented:
This is WIP, so don't complete the PR just yet. A few things need finishing/testing

* [x] Re-establish the Plugin mechanism - I commented it out as the whole Nuget.Core implementation has been deprecated & changed
* [x] There's no support for net framework - it's Net6 upwards - GE6 only support netcore currently
* [x] Complete the new 4x3 geometries / placement 
* [x] Reinstate code signing
* [x] I had to manually resolve a M.E.Logging.Abstractions version clash
* [ ] Project/package meta data needs a review as I had to remove the AssemblyInfos
* [ ] Builds need testing
* [x] Testing - lots of change here: new Helix core version, replacement AvalonDock, new GE, new Essentials, + netcore

andyward committed:
Added x86 as a target to support use on ARM64 until we get native support in GE
Tweaked About & assemblies tabs to make clearer what's loaded

andyward committed:
Updated latest netcore GE-6.1.552

andyward committed:
Show affected entity ID in logs
Helps with diagnostics where there may not be any additional context

andyward commented on Xbim.WindowsUI.sln:
```diff
@@ -29,29 +30,48 @@ EndProject
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
 		{89F98E53-DE4E-4427-8244-D5934AD08849}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{89F98E53-DE4E-4427-8244-D5934AD08849}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{89F98E53-DE4E-4427-8244-D5934AD08849}.Debug|x86.ActiveCfg = Debug|x86
```
x86 allows building/testing on Arm64 (until we get a GE & Xbim.Geometry.Occt with Arm64 support)

andyward commented on Xbim.WinformsSample/Properties/AssemblyInfo.cs:
```diff
@@ -22,15 +16,3 @@
 // The following GUID is for the ID of the typelib if this project is exposed to COM
 [assembly: Guid("55A780CE-22A2-434A-BA2D-A368CBEFD959")]
 
-// Version information for an assembly consists of the following four values:
```
we now use <GenerateAssemblyInfo> in the csproj

andyward committed:
Plugins system ported to net standard approach

Now looks for 6.0.0.0 compatible plugins as 4/5.x will not be compatible with v6

Breaking change: Plugins cannot be WinExe - these can't be assembly loaded in netcore: https://stackoverflow.com/a/58291394

TODO: consider multi-targetting
TODO: Review plugin commands

andyward committed:
Build on PRs

andyward committed:
Updated to latest DI / XbimServices & Logging
Updated to v6.0.0 xbim dependencies (and AssemblyVersions)

andyward committed:
Updated latest GE from feature/netcore
Includes SRL fix for Cylindrical surface in advanced breps

SteveLockley committed:
Updated geometry engine

andyward committed:
Updated latest GE

andyward committed:
Updated 3rd party dependencies

maxplex commented:
I'm really keen to check this out, is there an ETA for release?

CBenghi commented:
I have plans to pick this up next week.


On Tue, Jun 6, 2023, 02:55 maxplex ***@***.***> wrote:

> I'm really keen to check this out, is there an ETA for release?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/xBimTeam/XbimWindowsUI/pull/196#issuecomment-1577739633>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABJY7MP626JKZPXTYL3SBI3XJZ5XBANCNFSM6AAAAAATPBKMSA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

andyward committed:
Updated latest Geometry

andyward committed:
Update to latest v6 GE

andyward committed:
Updated latest dependencies (net6)

andyward committed:
Updated main app to .NET 8

CBenghi commented:
Hi @andyward,
I've tried to compile this branch, but it fails to restore the projects because of missing packages.

```
1>C:\Data\Dev\XbimLtd\XbimWindowsUI\Xbim.Presentation\Xbim.Presentation_isxv1we2_wpftmp.csproj : error NU1102: Unable to find package Xbim.ModelGeometry.Scene with version (>= 6.1.739-netcore)
1>C:\Data\Dev\XbimLtd\XbimWindowsUI\Xbim.Presentation\Xbim.Presentation_isxv1we2_wpftmp.csproj : error NU1102:   - Found 34 version(s) in xbim_master [ Nearest version: 5.1.541 ]
1>C:\Data\Dev\XbimLtd\XbimWindowsUI\Xbim.Presentation\Xbim.Presentation_isxv1we2_wpftmp.csproj : error NU1102:   - Found 11 version(s) in nuget.org [ Nearest version: 5.1.437 ]
1>C:\Data\Dev\XbimLtd\XbimWindowsUI\Xbim.Presentation\Xbim.Presentation_isxv1we2_wpftmp.csproj : error NU1102:   - Found 10 version(s) in xbim_develop [ Nearest version: 5.1.742-develop ]
```
Is there an easy fix for this other than compiling geometry locally?

I think that local management would require some hacking of the nuget package version, because geom packages are at v. 6.0.0 from a normal compilation.

andyward commented:
Claudio I'll invite you to our private nuget feed in Axure Devops. Separately will chat with the team about how we push to a public nuget (we wanted to hold off while it' still not a complete 4x3 implementation)

andyward committed:
Removed dependency on deprecated DotNetZip - use .net built in Compression.
Mainly used by Bcf plugin.
Optimised plugin startup, and added diagnostics setting to pluginConfig

andyward committed:
Updated latest netcore GE

andyward committed:
Updated BCF addin to .net8

andyward committed:
Updated dependencies

andyward committed:
Fix Typo in Measures

andyward committed:
Updated to latest GE with IFC4x3 linear placement, geometries etc.

andyward committed:
Updated latest Essentials

andyward committed:
Updated 3rd party dependencies

andyward committed:
Re-enabled code signing for those that want it

andyward committed:
Minor tidy/formatting

andyward committed:
Fixed up Drag and Drop plugin loading
Removed Extraneous plugin code.
Added Package Icon

andyward committed:
Run builds on feature branches / prs
